docs/internal-documentation/templates/specifications/tests/Template01.md

```diff
@@ -0,0 +1,268 @@
+  with no message tag outputted.
+* If a CDS record is of "delete" type, then it can by definition not
+  match or point at any DNSKEY record.
+
+> > Here is a table of all message tags referred to in the steps. The tag in the
+> > first column, the default severity level in the second, and a statement on
+> > when the message is outputted in the fourth.
+> >
+> > If data from the test, e.g. list of name server IP addresses, is to be
+> > outputted with the message, then the datatypes are listed in the third
+> > column, "arguments". The third column is left empty when no arguments are
+> > used for the message.
+> >
+> > Always use the same table set-up, but with the correct tags. Keep the table
+> > sorted by message tag. Here is an example:
+
+Message Tag outputted         | Level   | Arguments            | Description of when message tag is outputted
+:-----------------------------|:--------|:---------------------|:--------------------------------------------
+T01_ALGO_NOT_SUPPORTED_BY_ZM  | NOTICE  | algo_descr, algo_num | The algorithm used is not supported by the Zonemaster implementation.
+T01_BROKEN_DNSSEC             | ERROR   | ns_ip_list           | Replies do not follow the standard.
+T01_HAS_NSEC                  | INFO    |                      | The *Child Zone* uses NSEC.
+T01_HAS_NSEC3                 | INFO    |                      | The *Child Zone* uses NSEC3.
+T01_INCONSISTENT_DNSSEC       | ERROR   | keytag               | The configuration of the zone is inconsistent with respect to DNSSEC.
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+> > The message tags should be formed following the [Message Tag Specification].
+> >
+> > The default severity level is selected from the [Severity Level Definitions].
+
+## Test procedure
+
+> > This section contains the detailed steps to execute the test case. The steps
+> > should be as explicit as possible to avoid that different implementations or
+> > executions do different interpretations or assumptions.
+> >
+> > The steps should be written in such a way that it is reasonably possible to
+> > use them to execute the test case manually using tools such as [`dig`]. It
+> > can be assumed that the reader of the text has good understanding of DNS.
+> >
+> > The steps should state what messages (message tags) to be outputted when.
+> > Only messages with default severity level DEBUG or higher can be included.
+> >
+> > All messages with level INFO or higher must be included. I.e. the
+> > implementation should not include messages with default level INFO or higher
+> > unless included in the specification.
+> >
+> > Messages DEBUG or lower can be added in the implementation as needed.
```

Comment:
This statement isn't about how to write a good test procedure, but rather how to compare the equivalence between the specification and the implementation, right?

Reply:
Yes. I'll add a clarification.